linetools/spectra/lsf.py

```diff
@@ -364,17 +364,25 @@ def interpolate_to_wv_array(self,wv_array, kind='Akima'):
             # NT: I tried Akima interpolator from scipy.interpolate
             # and is not robust in extreme situations where the
             # wv_array is large compared to the kernel FWHM.
-            #Let's try linetools.analysis.interp Akima version 
+            # Let's try linetools.analysis.interp Akima version
             lsf_vals = interp_Akima(wv_array_AA,lsf_tab['wv'],lsf_tab['kernel'])
 
+        # make sure the kernel is never negative
+        cond = lsf_vals < 0
+        if np.sum(cond) > 0:
+            warnings.warn('The interpolated kernel has negative values; imposing them to be 0.')
+            if debug:
+                import matplotlib.pyplot as plt
+                plt.plot(wv_array_AA, lsf_vals, 'k-')
+                # import pdb; pdb.set_trace()
+            lsf_vals = np.where(lsf_vals < 0, 0., lsf_vals)
+
```

Comment:
I plotted up a kernel generated with interpolate_to_wv_array, and it seems to me that the wings approach zero too steeply (which likely caused the original function to go negative).  Did you check an interpolated version against the original LSF from STScI to ensure they line up?